cmake/CMakeLists.txt

```diff
@@ -350,6 +350,9 @@ if (onnxruntime_USE_ROCM)
   else()
     message(FATAL_ERROR "Cannot determine ROCm version string")
   endif()
+  if (ROCM_VERSION_DEV VERSION_GREATER_EQUAL "6.2")
```

Comment:
This need be inside `if (NOT CMAKE_HIP_ARCHITECTURES)` scope, otherwise, user specified CMAKE_HIP_ARCHITECTURES will not take effect.

Reply:
Thanks Tianlei!